cmd/vulcan-retirejs/main.go

```diff
@@ -99,77 +108,103 @@ func runRetireJs(ctx context.Context, args []string) ([]RetireJsFileResult, erro
-						Score: 0.0,
-						Recommendations: []string{
-							fmt.Sprintf("Check if there is an update available for the affected components."),
-							"Additional vulnerability information can be found in the links in resources",
-						},
-						Resources: []report.ResourcesGroup{
-							report.ResourcesGroup{
-								Name: "Vulnerable Components",
-								Header: []string{
-									"Component",
-									"Version",
-								},
-							},
+	for _, f := range r {
+		for _, v := range f.Results {
+			fingerprint := []string{}
+			vulnerability := report.Vulnerability{
+				Summary: "Vulnerabilities in JavaScript Dependencies",
+				CWEID:   1104,
+				Labels:  []string{"potential", "web", "retirejs"},
+				Description: "Vulnerabilities in dependencies may impact in the security of your program. For that reason " +
+					"it's important to check for issues not only in your code but in the 3rd party code you are using as a dependency.",
+				Recommendations: []string{
+					fmt.Sprintf("Check if there is an update available for the affected resource."),
+					"Additional vulnerability information can be found in the links in the resources table.",
+				},
+				References:       []string{"https://portswigger.net/kb/issues/00500080_vulnerable-javascript-dependency"},
+				AffectedResource: fmt.Sprintf("%s-%s", v.Component, v.Version), // example: jquery-1.9.0
+				Score:            0.0,
+				Resources: []report.ResourcesGroup{
+					{
+						Name: "Vulnerabilities",
+						Header: []string{
+							"Dependency",
+							"CVEs",
+							"Affected Versions",
+							"Severity",
+							"References",
 						},
-					}
+					},
+				},
+			}
+			details := []string{fmt.Sprintf("The following vulnerabilities were found in %s version %s JavaScript dependency:", v.Component, v.Version)}
+			fingerprint = append(fingerprint, fmt.Sprintf("vulnerabilities#%d", len(v.Vulnerabilities)))
+			for _, i := range v.Vulnerabilities {
+				if score := getScore(i.Severity); vulnerability.Score < score {
+					vulnerability.Score = score
 				}
-
-				if score := getScore(vuln.Severity); v.Score < score {
-					v.Score = score
+				fingerprint = append(fingerprint, strings.ToLower(i.Severity))
+				if i.Identifiers.Bug != "" {
+					fingerprint = append(fingerprint, i.Identifiers.Bug)
 				}
-
-				v.References = append(v.References, vuln.Info...)
-
-				gr := v.Resources[0]
+				if i.Identifiers.Issue != "" {
+					fingerprint = append(fingerprint, i.Identifiers.Issue)
+				}
+				details = append(details, fmt.Sprintf("- [%s] %s", strings.Join(i.Identifiers.Cve, ","), i.Identifiers.Summary))
+				references := ""
+				for i, reference := range i.Info {
+					if i > 0 {
+						references += ", "
+					}
+					references += fmt.Sprintf("[%d](%s)", i, reference)
+				}
+				gr := vulnerability.Resources[0]
 				r := map[string]string{
-					"Component": result.Component,
-					"Version":   result.Version,
+					"Dependency":        vulnerability.AffectedResource,
```

Comment:
As mentioned above: do you think this field could be removed?

Reply:
Done!